src/extensions/default/analyzers/Microsoft.DotNet.UpgradeAssistant.Extensions.Default.CSharp.Analyzers/AnalyzerExtensions.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Microsoft.DotNet.UpgradeAssistant.Extensions.Default.CSharp.Analyzers
+{
+    public static class AnalyzerExtensions
+    {
+        public static bool IsCorrectLanguage(this DiagnosticAnalyzer analyzer, Compilation comp)
+        {
+            if (analyzer is null)
+            {
+                return false;
+            }
+
+            var langFromAttributes = GetSpecifiedLanguage(analyzer.GetType());
+            if (!langFromAttributes.Any())
+            {
+                // this analyzer is language agnostic because it does not specify an attribute
+                return true;
+            }
+
+            return langFromAttributes.Any(lang => lang.Equals(comp.Language, StringComparison.Ordinal));
+        }
+
+        /// <summary>
+        /// Review the Attributes of the <paramref name="analyzer"/> to determine
+        /// if which languages apply.
+        /// </summary>
+        /// <param name="analyzer">True</param>
+        /// <returns>A list of supported languages.</returns>
+        private static IEnumerable<string> GetSpecifiedLanguage(Type analyzer)
+        {
+            var analyzerAttr = analyzer.CustomAttributes.FirstOrDefault(a => a.AttributeType.FullName.Equals(typeof(DiagnosticAnalyzerAttribute).FullName, StringComparison.Ordinal));
```

Comment:
This may be a very expensive operation if done within an analyzer. What scenario would this be used in?

Reply:
actually, we don't need this entire file. I had built this to create a guard clause. While running our unit tests we were loading both VB and C# child class analyzers which don't exist anymore. I'll remove the file.

In hindsight, I'm not sure why I used reflection here instead of a constant in the child class.